src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -148,16 +174,39 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             }
         }
 
-        private string GetDataContentType(JsonElement json, out bool isJson)
+        private void ForwardCloudEventPropertiesAsHeaders(
+            HttpContext httpContext,
+            IEnumerable<JsonProperty> jsonPropNames)
         {
+            if (!options.ForwardCloudEventPropertiesAsHeaders)
+            {
+                return;
+            }
+
+            foreach (var jsonProperty in jsonPropNames
+                         .Where(d => !ExcludedPropertiesFromHeaders.Contains(d.Name.ToLowerInvariant())))
+            {
+                httpContext.Request.Headers.TryAdd($"Cloudevent.{jsonProperty.Name.ToLowerInvariant()}",
```

Comment:
Is it certain that all string-ified JSON values are valid HTTP header values?  (I.e. are there characters, for example, that might exist in JSON that are not allowed in headers?) If not, it might strengthen the argument for allowing more fine-grained forwarding.

Reply:
This is a very interesting observation.
In our services I have been using a similarly modified middleware to forward them, and it has been working just fine, but they were plain utf8 text, however after doing a quick search some special characters seem to not be accepted in HTTP headers.
I wonder if that validation happens earlier or later in the middleware pipeline, which could affect this or not
that said, I would be concerned if engineers posted unencoded special characters in JSON payloads in the first place, as some brokers like Kafka or RabbitMQ etc might not support them.

More fine grained control on which props we should forward is definitely a good idea, I'll refactor it tomorrow

Reply:
@philliphoff 
How about this proposal:
I'm thinking of adding another flag to optionally URL encode the values that will be put in the headers, to allow people pass around non-ASCI characters if they need to.

Reply:
After some more reading it seems that AspNetCore URL encodes/decodes automatically UTF-8 characters that are not ASCI, I will add some tests to try it out just in case we need to explicitly encode the values

Reply:
I can confirm that non ASCI characters are forwarded just fine in the headers an the body without any changes required
![image](https://github.com/dapr/dotnet-sdk/assets/17237280/dfd0ed94-9eb3-4d73-b227-08d00121434c)
However I think two optional filter lists could be beneficial so ill proceed and add them as well